subtests/docker_cli/deferred_deletion/deferred_deletion.py

```diff
@@ -0,0 +1,145 @@
+between systems with (1) enabled incorrectly and correctly. We do
+so by checking that the sysctl knob /proc/sys/fs/may_detach_mounts
+(introduced in kernel 3.10.0-632) exists and is 1.
+
+We abort with TestNAError if any precondition fails.
+
+Note that we need to do all the important work from inside a
+helper script, because our autotest process will only have access
+to the container's mounts if MountFlags=slave is *absent* from
+the docker systemd unit file; by default that option is present,
+and it is not clear if/when it will be removed. To make sure
+we have access to the container's mounts we need to nsenter
+the docker daemon's mount namespace, and we can't do that within
+autotest (using ctypes.CDLL and setns) because autotest is
+multithreaded and setns() doesn't allow that.
+
+Operational Summary
+----------------------
+
+#. Run a docker container
+#. Run a test script inside docker mount space that will:
+#. - Find out the local (host) mountpoint of the container's root filesystem.
+#. - cd into that directory
+#. - Read the Deferred-Deletion count from 'docker info'.
+#. - Remove the container.
+#. - Read the Deferred-Deletion count; confirm that it has grown by one.
+#. - cd back out of the container rootfs
+#. - Read the Deferred-Deletion count; confirm that it has gone back down.
+
+"""
+
+import os
+from autotest.client import utils
+import dockertest.docker_daemon
+from dockertest import subtest
+from dockertest.containers import DockerContainers
+from dockertest.dockercmd import AsyncDockerCmd
+from dockertest.images import DockerImage
+from dockertest.output.validate import mustpass
+from dockertest.xceptions import DockerTestNAError
+
+
+class deferred_deletion(subtest.Subtest):
+
+    def initialize(self):
+        super(deferred_deletion, self).initialize()
+
+        docker_cmdline = dockertest.docker_daemon.cmdline()
+        self.failif_not_in("dm.use_deferred_deletion=true",
+                           str(docker_cmdline),
+                           "docker daemon command-line options",
+                           DockerTestNAError)
+
+        self._rhel72_bug_workaround()
+
+        self.stuff['dc'] = DockerContainers(self)
+
+    def _rhel72_bug_workaround(self):
+        """
+        On RHEL, docker daemon may be running with deferred_deletion
+        enabled incorrectly.
+        """
+
+        try:
+            self.failif_not_redhat()
+        except DockerTestNAError:
+            return                   # Fedora or CentOS
+
+        # RHEL
+        sysctl_knob = "/proc/sys/fs/may_detach_mounts"
+        self.failif(not os.path.exists(sysctl_knob),
+                    "sysctl knob %s does not exist; this system is"
+                    " not likely to support deferred deletion" % sysctl_knob,
+                    DockerTestNAError)
+
+        with open(sysctl_knob, 'rb') as fs_may_detach_mounts:
+            enabled = int(fs_may_detach_mounts.read().strip())
+            self.failif(not enabled,
```

Comment:
optional: Can ``failif_ne()`` be used for these checks at all?  Sometimes these knobs end up having more values added on later.  Not very likely, mostly thinking that the output from that function may save some error-message writing (here).